File: node_api/models/Doctor.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const DoctorSchema = new Schema({
    name: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true
    },
    password: {
        type: String,
        required: true
    },
    phone: {
        type: String
    },
    avatar: {
        type: String,
        default: 'default avatar path here'
    },
    city: {
        type: String,
        required: true
    },
    clinickAddress: {
        type: String,
        required: true
    },
    gender: {
        type: String,
    },
    specialization: {
        type: String,
        required: true
    },
    canTreatFor: [
        {
            disease: {
                type: String
            }
        }
    ],
    openingTime: {        //type is taken string because there is no shema type avialable for time
        type: String
    },
    closingTime: {
        type: String
    }
});

module.exports = Doctor = mongoose.model('doctors', DoctorSchema);

File: medic_flutter_app/lib/models/doctor.dart
import 'package:flutter/cupertino.dart';

class Doctor {
  final String name;
  final String userId;
  final String email;
  final String phone;
  final String avatar;
  final int age;
  final List<dynamic> reports;
  final String address;
  final String city;
  final String gender;
  final String token;
  Doctor({
    @required this.name,
    @required this.userId,
    @required this.email,
    @required this.phone,
    @required this.avatar,
    @required this.reports,
    @required this.age,
    @required this.address,
    @required this.gender,
    @required this.city,
    this.token,
  });
}
